07_modules_libs/homework/stock/stock.go

```diff
@@ -0,0 +1,105 @@
+package stock
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type StockStat struct {
+	Ticker       string  `json:"ticker"`
+	HighestPrice float64 `json:"highest_price,string"`
+	LowestPrice  float64 `json:"lowest_price,string"`
+	AvgPrice     float64 `json:"avg_price,string"`
+}
+
+func GetStockStat(stockTicker string, apiKey string) (*StockStat, error) {
+	url := fmt.Sprintf(
+		"https://www.alphavantage.co/query?function=OVERVIEW&symbol=%v&apikey=%v",
+		strings.ToUpper(stockTicker), apiKey,
+	)
+	response, err := http.Get(url)
+	if err != nil {
+		return &StockStat{}, err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		errorMessage, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return &StockStat{}, err
```

Comment:
Common pattern is returning `nil, err`. No need to allocate memory in case of error.

Reply:
Exactly! I forgot I can return nil instead of a pointer